app/src/main/java/com/example/bigowlapp/utils/AuthenticatorByPhoneNumber.java

```diff
@@ -0,0 +1,66 @@
+package com.example.bigowlapp.utils;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.telephony.TelephonyManager;
+import android.widget.Toast;
+
+import com.example.bigowlapp.model.Attendance;
+import com.example.bigowlapp.model.AuthByPhoneNumberFailure;
+import com.example.bigowlapp.model.Schedule;
+import com.example.bigowlapp.model.User;
+import com.example.bigowlapp.model.UserScheduleResponse;
+import com.example.bigowlapp.repository.AuthRepository;
+import com.example.bigowlapp.repository.RepositoryFacade;
+import com.google.firebase.Timestamp;
+
+public class AuthenticatorByPhoneNumber {
+
+    private String deviceIdNumber;
+    private String devicePhoneNumber;
+    private String currentUserPhoneNumber;
+    private final Context context;
+    private final AuthRepository authRepository = new AuthRepository();
+
+    RepositoryFacade repositoryFacade = RepositoryFacade.getInstance();
+
+    public AuthenticatorByPhoneNumber(Context context) {
+        this.context = context;
+    }
+
+    @SuppressLint("MissingPermission")
+    public void authenticate(String scheduleId) {
+        TelephonyManager telephonyManager = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
+        deviceIdNumber = telephonyManager.getDeviceId();
```

Comment:
- the READ_PRIVILEGED_PHONE_STATE is actually still needed, since the getDeviceId() call uses it
- "getDeviceId()" is always null on Android 10 plus devices and also causes a crash, which can be fixed by surrounding it with a try/catch. Also if it being null is an issue, you might want to use other methods as I had recommended in my previous review.

Reply:
I have placed a condition for the "getDeviceId()" to be called only for Android versions less than 10. For versions equal and greater than 10, another method will be used

Reply:
This is no more relevant as I decided to go with the app instance ID in place of the device ID. That solution is less troublesome and is the one most recommended for best practice for unique identifier in this case